### PR TITLE
Add 'rand' uniform-random placer.

### PR DESCRIPTION
**Pull request by mossblaser**
This placer is intended as a baseline for comparing placement algorithms. It
places each vertex on a random core (not sequentially filling random chips like
the sa placer with effort=0.0) and respects all constraints.

### CODE REVIEW

File: rig/place_and_route/place/rand.py [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+            apply_reserve_resource_constraint(machine, constraint)
+
+    # The set of vertices which have not been constrained.
+    movable_vertices = [v for v in vertices_resources
+                        if v not in placements]
+
+    locations = set(machine)
+
+    for vertex in movable_vertices:
+        # Keep chosing random chips until we find one where the vertex fits.
```
Comment (mundya) [inline rig/place_and_route/place/rand.py]: chosing -> choosing

File: rig/place_and_route/place/rand.py [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+
+    This algorithm performs uniform-random placement of vertices (completely
+    ignoring connectivty) and thus in the general case is likely to produce
+    very poor quality placements. It exists primarily as a baseline comparison
+    for placement quality and is probably of little value to most users.
+
+    Parameters
+    ----------
+    random : :py:class:`random.Random`
+        A Python random number generator. Defaults to ``import random`` but can
```
Comment (mundya) [inline rig/place_and_route/place/rand.py]: While I'm being picky... not sure the first sentence is necessary

**Comment by mundya** [comment]
LGTM, feel free to ignore my incredibly picky reviewing!

**Comment by mossblaser** [comment]
Thanks very much for that; @mundya!

**Comment by mundya** [comment]
No problem - feel free to merge after the tests.

Rigged

**Pull request merged**